lib/lhm/chunk_insert.rb

```diff
@@ -0,0 +1,41 @@
+module Lhm
+  class ChunkInsert
+    def initialize(migration, lowest, highest)
+      @migration = migration
+      @router = Router.new(migration)
+      @lowest = lowest
+      @highest = highest
+    end
+
+    def insert(connection)
+      connection.insert(sql)
```

Comment:
My reading is that this insert is going to go to `ActiveRecord::ConnectionAdapter::DatabaseStatements` -- in MySQL because of https://github.com/rails/rails/blob/8f3178f3132f618d982e71554e192f6282fcb159/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb#L125 that is going to cause an extra round-trip to the server to do `SELECT LAST_INSERT_ID()`.

Even though these are clearly `INSERT`s, I still think you actually want to call `connection.update(sql)`, for that reason.